packages/@aws-cdk/aws-route53/lib/records/alias.ts

```diff
@@ -0,0 +1,53 @@
+import { CloudFrontWebDistribution } from '@aws-cdk/aws-cloudfront';
+import { BaseLoadBalancer } from '@aws-cdk/aws-elasticloadbalancingv2';
+import { Construct } from '@aws-cdk/cdk';
+import { HostedZoneRef } from '../hosted-zone-ref';
+import { cloudformation } from '../route53.generated';
+import { determineFullyQualifiedDomainName } from './_util';
+
+export interface AliasRecordProps {
+  hostedZone: HostedZoneRef | string;
```

Comment:
I think Elad mentioned this on another PR, but we don't really support type unions even though we don't prohibit their use.

Just the `HostedZoneRef` should be sufficient, no?

if you really *really* care about strings here, make a separate property for the string variant and add a runtime check that exactly one of them should be supplied.

Reply:
Alternative would be to make hostedZoneId optional in HostedZoneRef, but I'd need to check if hostedZoneId is optional everywhere.  I only know it is for RecordSets

Reply:
I'm trying to avoid having to put the zone ID in a file, even if it's a context file. It's unnecessary, since I should be able to produce a template with the name only

Reply:
It's making API calls to Route53 to retrieve zone IDs given a name.

Yes, the ID also ultimately ends up in a file but since you don't have to put it there, I don't see why you would care?